addons/website_slides/static/src/js/activity.js

```diff
@@ -42,4 +42,178 @@ function applyInclude(Activity) {
 
 applyInclude(KanbanActivity);
 
-});
\ No newline at end of file
+});
+
+odoo.define('website_slides/static/src/js/slide_models.js', function (require) {
+    'use strict';
+
+    const {
+        registerClassPatchModel,
+        registerFieldPatchModel,
+        registerNewModel
+    } = require('mail/static/src/model/model_core.js');
+    const { attr, one2one } = require('mail/static/src/model/model_field.js');
+
+    registerClassPatchModel('mail.activity', 'website_slides/static/src/js/slide_models.js', {
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        convertData(data) {
+            const data2 = this._super(data);
+            if ('request_partner_id' in data && 'res_id' in data) {
+                if (!data.request_partner_id) {
+                    data2.slide = [['unlink-all']];
+                } else {
+                    data2.slide = [
+                        ['insert', {
+                            id: data.res_id,
+                            partnerId: data.request_partner_id[0],
+                        }],
+                    ];
+                }
+            }
+            return data2;
+        },
+    });
+
+    registerFieldPatchModel('mail.activity', 'website_slides/static/src/js/slide_models.js', {
+        slide: one2one('slides.slide', {
+            inverse: 'activity',
+        }),
+    });
+
+    function factory(dependencies) {
+
+        class Slide extends dependencies['mail.model'] {
+            //----------------------------------------------------------------------
+            // Public
+            //----------------------------------------------------------------------
+
+            /**
+             * Grant the access for the current `Course`.
+             */
+            async grantAccess() {
+                await this.async(() => this.env.services.rpc({
+                    model: 'slide.channel',
+                    method: 'action_grant_access',
+                    args: [this.id, this.partnerId],
+                }));
+                if (this.activity) {
+                    this.activity.delete();
+                }
+                this.delete();
+            }
+            /**
+             * Refuse the access for the current `Course`.
+             */
+            async refuseAccess() {
+                await this.async(() => this.env.services.rpc({
+                    model: 'slide.channel',
+                    method: 'action_refuse_access',
+                    args: [this.id, this.partnerId],
```

Comment:
prefer always passing a list of id instead of single id
and prefer sending named params (here partner_id) as kwargs